src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/custom.py

```diff
@@ -38,28 +57,46 @@ def register_cloud(cloud_name,
                    suffix_keyvault_dns=None,
                    suffix_azure_datalake_store_file_system_endpoint=None,
                    suffix_azure_datalake_analytics_catalog_and_job_endpoint=None):
-    if cloud_config:
-        # Using JSON format so convert the keys to snake case
-        for key in cloud_config:
-            cloud_config[to_snake_case(key)] = cloud_config.pop(key)
-        cloud_args = cloud_config
-    else:
-        cloud_args = locals()
-    c = Cloud(cloud_name)
-    for arg in cloud_args:
-        if arg.startswith('endpoint_') and cloud_args[arg]:
-            setattr(c.endpoints, arg.replace('endpoint_', ''), cloud_args[arg])
-        elif arg.startswith('suffix_') and cloud_args[arg]:
-            setattr(c.suffixes, arg.replace('suffix_', ''), cloud_args[arg])
+    c = _build_cloud(cloud_name, cloud_config=cloud_config,
+                     cloud_args=locals())
     try:
         add_cloud(c)
     except CloudAlreadyRegisteredException as e:
         raise CLIError(e)
 
+def modify_cloud(cloud_name=None,
+                 cloud_config=None,
+                 endpoint_management=None,
```

Comment:
Can we accept the a list of `key-value` pair, rather using strong named arguments? By doing that, we will have a little leeway to allow some service teams to introduce new end-points such as temporary test endpoints